packages/strapi/lib/core/app-configuration/config-loader.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+const { templateConfiguration } = require('strapi-utils');
+
+const env = require('./env-helper');
+
+module.exports = dir => {
+  if (!fs.existsSync(dir)) return {};
```

Comment:
It could be clearer to return undefined in order to differentiate empty conf & file not found.
imo, it should be the caller job to handle default value (such as `{}`), not the callee.

Reply:
This isn't about default values here but about type consistency. 
We can define the config-loader function as (string) -> Object. 

Returning undefined would break this definition and force a user to handle different return values instead of just handling an object or an exception.

If we were do consider to handle empty conf we would handle it via exceptions and errors not via return types.

Reply:
Although I got your point, I'm not sure hiding errors behind a default value is a good practice. So if doing `(string)?: Object` cannot be considered, throwing an exception could be a great compromise. 🙂 

PS: I might be influenced too much by TS since, based on the type checking, it's easy to spot where you don't handle potentially undefined values, which is not necessarily true in JS.

Reply:
As Discussed more on slack :) For those interesed, In short. 

This is not considered hiding an error but just a default behavior for convenience sake (e.g rm -rf doesn't throw if folder doesn't exist) 

It would be better to do the existence check before calling the function or use a Maybe monad instead of handling undefined. Using a zero value is pretty common to :)